### PR TITLE
add auth rfc

### PR DESCRIPTION
**Pull request by israx**

### CODE REVIEW

File: RFC.md [hunk 1]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
 
 # `Auth` Changes
 
-- CognitoUser object is no longer required
+Amplify is proposing the following changes for the `Auth` category.
 
-Code sample for Amplify V5 (`aws-amplify@5`)
+## Typescript support for user attributes
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
## TypeScript support for user attributes
```

File: RFC.md [hunk 2]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
...
-  // done
-}
+```Typescript
+
+  Auth.signUp({
+	  username:"username",
+      password:"*******",
+	  attributes:{
+		email: "email@email"
+	 }})
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
```Typescript
Auth.signUp({
  username: "username",
  password: "*******",
  attributes: {
    email: "email@domain.com"
  }
});
```

File: RFC.md [hunk 3]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
...
+```Typescript
+
+  signUp({
+    username: "username",
+    password: "********",
+    options: {
+      userAttributes: [{ userAttributeKey: "email", value: "email@email.com" }],
+    },
+  })
+
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
signUp({
  username: "username",
  password: "********",
  options: {
    userAttributes: [{ userAttributeKey: "email", value: "email@domain.com" }],
  },
});
```

File: RFC.md [hunk 4]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
...
+```
+
+**Proposed Amplify v6 DX**
+
+Todo: add screen shoots
+
+## Predictable API responses
+
+We are improving **_DX_** by providing descriptive API responses that help customers to complete auth flows.
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
We are improving **_DX_** by providing descriptive API responses to help customers complete auth flows. An example for the `confirmSignUp` API is highlighted below.
```

File: RFC.md [hunk 5]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
...
+
+**Amplify v5 (`aws-amplify@5`)**
+
+```Typescript
+
+const resp = Auth.confirmSignUp("username", "112233")
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
const resp = Auth.confirmSignUp("username", "112233");
```

File: RFC.md [hunk 6]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
...
+
+  const resp = await confirmSignUp({
+    username: "username",
+    confirmationCode: "112233",
+  });
+
+  if (resp.isSignUpComplete) {
+    // Show login component
+  }
+
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
const resp = await confirmSignUp({
  username: "username",
  confirmationCode: "112233",
});
```

Removed the `resp.isSignUpComplete` part from my suggestion, but I'm okay with it as long as we also show the equivalent flow for v5 (consistency between samples).

File: RFC.md [hunk 7]
```diff
@@ -147,49 +147,84 @@ Try out the new types here: TODO Playground Link
 
 # `Auth` Changes
 
-- CognitoUser object is no longer required
+Amplify is proposing the following changes for the `Auth` category.
```
Comment (jimblanc) [inline RFC.md]:
```suggestion
Amplify is proposing the following changes for the `Auth` category. Similar changes will be applied across all of the `Auth` APIs but examples for specific APIs are highlighted below.
```

Added some wording to make it clear these changes will be applied across auth (which customers can experiment with in the playground)

**Review by jimblanc — COMMENTED** [review]
Some minor feed back but looks good! It's tricky with Auth due to the scope of changes, but I think you've selected some good examples.

File: RFC.md (hunk 6 above)
Comment (israx) [reply RFC.md]: just added a similar flow to v5

**Pull request merged**